src/Entities/Entity.cpp

```diff
@@ -1451,7 +1450,7 @@ bool cEntity::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 	}
 
 	// Remove all links to the old world
-	SetWorldTravellingFrom(GetWorld());  // cChunk::Tick() handles entity removal
+	GetWorld()->GetChunkMap()->RemoveEntity(this);
```

Comment:
I believe the world tick thread and whatever thread lets plugins run binding code.